src/lib/caches/Items.js

```diff
@@ -94,22 +100,74 @@ module.exports = class ItemsCache {
       let exact = false;
       filtered = undefined;
       base
-        .filter((item) => item && item[by])
+        .filter((item) => item && this.#parseNestedKey(item, bys))
         .forEach((item) => {
-          if (item[by].toLowerCase() === term.toLowerCase()) {
-            filtered = item;
-            exact = true;
+          const values = this.#parseNestedKey(item, bys);
+          // parseNestedKey returns an array, so check all values in that array
+          for (const entry of values) {
+            //logger.error(entry.toLowerCase());
+            if (entry.toString().toLowerCase() === term.toString().toLowerCase()) {
+              filtered = item;
+              exact = true;
+            }
           }
 
-          if (item[by].toLowerCase().includes(term.toLowerCase()) && !exact) {
-            const distance = item[by].toLowerCase().replace(term.toLowerCase(), '').length;
-            if (!keyDistance || distance < keyDistance) {
-              keyDistance = distance;
-              filtered = item;
+          if (!exact) {
+            // parseNestedKey returns an array, so check all values in that array
+            for (const entry of values.values()) {
+              if (entry.toString().toLowerCase().includes(term.toString().toLowerCase())) {
+                const distance = entry.toString().toLowerCase().replace(term.toString().toLowerCase(), '').length;
+                if (!keyDistance || distance < keyDistance) {
+                  keyDistance = distance;
+                  filtered = item;
+                }
+              }
             }
           }
         });
     }
     return this.#cleanup(filtered, { remove, only });
   }
+
+  /**
+   * Utility function for following nested keys. Since some keys have values that are
+   * an array of non-keyed objects, this function iterates over those arrays and can
+   * return multiple values for a single key path.
+   * @param {Array} item data to search
+   * @param {Array<string>} by array of keys to follow
+   * @returns {Array<string>} or undefined if an invalid path of keys is provided
+   */
+  static #parseNestedKey(item, by) {
+    let resultList = [];
+
+    // Walk through keys in order
+    by.reduce((prevNode, currentValue, index) => {
+      if (!prevNode) {
+        return undefined;
+      }
+      // If an entry isn't found with the key, check for an unkeyed array.
+      // This allows a user to use 'by=components.name' rather than 'by=components.0.name'
+      // and find every entry that path might meet.
+      if (!prevNode[currentValue]) {
+        if (Array.isArray(prevNode)) {
+          resultList = Array.from(prevNode, (element) => this.#parseNestedKey(element, by.slice(index)))
+                            .flat()
```

Comment:
```suggestion
                            .flat(Infinity)
```

this forces any level of dept that we introduce to get flattened